junit-platform-engine/src/main/java/org/junit/platform/engine/support/hierarchical/ResourceLock.java

```diff
@@ -43,4 +43,12 @@ default void close() {
 		release();
 	}
 
+	/**
+	 * {@return whether the given lock is compatible with this lock}
+	 * @param other the other lock to check for compatibility
+	 */
+	default boolean isCompatible(ResourceLock other) {
+		return other instanceof NopLock || this instanceof NopLock;
```

Comment:
nit: We should probably switch this, as the `this` case is more frequent.

Reply:
Done in cd364fc3c3fb4d33f6b946964912166950588718.